Source/DafnyRuntime/DafnyRuntimeJava/src/main/java/dafny/DafnyMap.java

```diff
@@ -29,16 +29,16 @@ public DafnyMap(Map<K, V> m) {
     }
 
     @SuppressWarnings("unchecked")
-    public static <K, V> Type<DafnyMap<K, V>> _type(
+    public static <K, V> Type<DafnyMap<? extends K, ? extends V>> _type(
             Type<K> keyType, Type<V> valueType) {
         // Fudge the type parameters; it's not great, but it's safe because
         // (for now) type descriptors are only used for default values
-        return Type.referenceWithInitializer(
-                (Class<DafnyMap<K, V>>) (Class<?>) DafnyMap.class,
-                DafnyMap::empty);
+        return Type.referenceWithDefault(
+                (Class<DafnyMap<? extends K, ? extends V>>) (Class<?>) DafnyMap.class,
```

Comment:
nit: there's no point in initially casting to `(Class<?>)`: `DafnyMap.class` will already have type `Class<DafnyMap>`, which is a strict subtype of `Class<?>`.